services/checks/registry_test.go

```diff
@@ -83,4 +83,197 @@ func TestRegistry(t *testing.T) {
 		require.Equal(t, 1, cap(collectedAlerts))
 		assert.Equal(t, expectedAlert, collectedAlerts[0])
 	})
+
+	t.Run("delete check result", func(t *testing.T) {
+		alertTTL := resolveTimeoutFactor * defaultResendInterval
+		r := newRegistry(alertTTL)
+
+		nowValue := time.Now().UTC().Round(0) // strip a monotonic clock reading
+		r.nowF = func() time.Time { return nowValue }
+		checkResults := []sttCheckResult{
+			{
+				checkName: "name1",
+				target: target{
+					agentID:   "/agent_id/123",
+					serviceID: "/service_id/123",
+					labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				result: check.Result{
+					Summary:     "check summary 1",
+					Description: "check description 1",
+					ReadMoreURL: "https://www.example.com",
+					Severity:    common.Warning,
+					Labels: map[string]string{
+						"baz": "qux",
+					},
+				},
+			},
+			{
+				checkName: "name2",
+				target: target{
+					agentID:   "/agent_id/321",
+					serviceID: "/service_id/321",
+					labels: map[string]string{
+						"bar": "foo",
+					},
+				},
+				result: check.Result{
+					Summary:     "check summary 2",
+					Description: "check description 2",
+					ReadMoreURL: "https://www.example2.com",
+					Severity:    common.Notice,
+					Labels: map[string]string{
+						"qux": "baz",
+					},
+				},
+			},
+		}
+
+		r.set(checkResults)
+		r.delete([]string{"name1"})
+
+		expectedAlert := &ammodels.PostableAlert{
+			Annotations: map[string]string{
+				"summary":       "check summary 2",
+				"description":   "check description 2",
+				"read_more_url": "https://www.example2.com",
+			},
+			EndsAt: strfmt.DateTime(nowValue.Add(alertTTL)),
+			Alert: ammodels.Alert{
+				Labels: map[string]string{
+					"alert_id":  "/stt/8fa5695dc34160333eeeb05f00bf1ddbd98be59c",
+					"alertname": "name2",
+					"qux":       "baz",
+					"bar":       "foo",
+					"severity":  "notice",
+					"stt_check": "1",
+				},
+			},
+		}
+
+		collectedAlerts := r.collect()
+		require.Len(t, collectedAlerts, 1)
+		require.Equal(t, 1, cap(collectedAlerts))
```

Comment:
why checking capacity?

Reply:
It was copied from another test. Removed everywhere.